api/src/main/java/org/openmrs/module/reporting/dataset/definition/evaluator/IterableSqlDataSetEvaluator.java

```diff
@@ -0,0 +1,142 @@
+import org.openmrs.module.reporting.dataset.DataSet;
+import org.openmrs.module.reporting.dataset.DataSetColumn;
+import org.openmrs.module.reporting.dataset.IterableSqlDataSet;
+import org.openmrs.module.reporting.dataset.definition.DataSetDefinition;
+import org.openmrs.module.reporting.dataset.definition.IterableSqlDataSetDefinition;
+import org.openmrs.module.reporting.dataset.definition.SqlFileDataSetDefinition;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.common.SqlIterator;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.util.OpenmrsUtil;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * The logic that evaluates a {@link IterableSqlDataSetDefinition} and produces an {@link DataSet}
+ *
+ * @see IterableSqlDataSetDefinition
+ */
+@Handler(supports = {IterableSqlDataSetDefinition.class})
+public class IterableSqlDataSetEvaluator implements IterableDataSetEvaluator {
+
+    protected Log log = LogFactory.getLog(this.getClass());
+
+    /**
+     * Public constructor
+     */
+    public IterableSqlDataSetEvaluator() {
+    }
+
+    /**
+     * @should evaluate a IterableSqlDataSetDefinition
+     * @should evaluate a IterableSqlDataSetDefinition with parameters
+     * @should evaluate a IterableSqlDataSetDefinition with in statement
+     * @should protect SQL Query Against database modifications
+     * @see DataSetEvaluator#evaluate(DataSetDefinition, EvaluationContext)
+     */
+    public DataSet evaluate(DataSetDefinition dataSetDefinition, EvaluationContext context) throws EvaluationException {
+
+        context = ObjectUtil.nvl(context, new EvaluationContext());
+
+        IterableSqlDataSetDefinition sqlDsd = (IterableSqlDataSetDefinition) dataSetDefinition;
+
+        SqlQueryBuilder queryBuilder = new SqlQueryBuilder();
+
+        String sqlQuery = sqlDsd.getSql();
+
+        queryBuilder.append(sqlQuery);
+
+        queryBuilder.setParameters(context.getParameterValues());
+
+        IterableSqlDataSetDefinition defenition = (IterableSqlDataSetDefinition) dataSetDefinition;
+        Properties connectionProperties = getConnectionProperties(defenition.getConnectionPropertyFile());
+        Iterator iterator = null;
+        Connection connection = null;
+        try {
+            connection = createConnection(connectionProperties);
+            SqlRunner runner = new SqlRunner(connection);
+
+            if (StringUtils.isNotBlank(defenition.getSql())) {
+                iterator = runner.executeSqlToIterator(defenition.getSql());
+
+            }
+
+        } catch (EvaluationException ee) {
```

Comment:
Is there any value added by this catch?

Reply:
The goal of this catch is to not encapsulate an `Exception` that already is `EvaluationException`.

Reply:
I do not understand the above.

Reply:
The method  `evaluate()` is throwing an `EvaluationException`. 
And any other exception is being converted for an `EvaluationException`. 
So if is already an `EvaluationException` doesent need to be converted.

Reply:
What do we lose if we remove this catch for `EvaluationException`?

Reply:
We may end up having `EvaluationException` that in the stack trace will have another `EvaluationException`, creating some mor complexity analysing the error.
It's not a big deal, I just think that is more clean this way.